packages/ui/src/models/visualization/flows/pipe-visual-entity.ts

```diff
@@ -73,11 +97,17 @@ export class PipeVisualEntity implements BaseVisualCamelEntity {
     return this.spec;
   }
 
-  updateModel(path: string | undefined, value: unknown): void {
+  updateModel(path: string | undefined, value: Record<string, unknown>): void {
     if (!path) return;
 
-    const stepModel = get(this.spec, path) as PipeStep;
-    if (stepModel) set(stepModel, 'properties', value);
+    if (path === ROOT_PATH) {
+      updatePipeFromCustomSchema(this.metadata, value);
+      this.id = this.metadata.name as string;
+      return;
+    }
+
+    const stepModel = getValue(this.spec, path) as PipeStep;
+    if (stepModel) setValue(stepModel, 'properties', value);
```

Comment:
we had this approach with `Kamelets` and this cause for the metadata and the visual entity to go out of sync, I think this should be consolidated into a single object, similar to this constructor: https://github.com/KaotoIO/kaoto/blob/4ef92c9ef261a72027aad9805beab39791195076/packages/ui/src/models/visualization/flows/kamelet-visual-entity.ts#L24-L29